keras_cv/models/segmentation/segment_anything/sam_layers.py

```diff
@@ -278,7 +278,7 @@ def __init__(self, num_positional_features, scale, **kwargs):
         self.positional_encoding_gaussian_matrix = self.add_weight(
             name="positional_encoding_gaussian_matrix",
             shape=(2, self.num_positional_features),
-            dtype=self.dtype,
+            dtype=self.variable_dtype,
```

Comment:
this line should be unnecessary I think, `add_weight` will always default to the variable dtype

Reply:
Sounds good, thanks for clarifying!